aspnet/mvc/models/file-uploads.rst

```diff
@@ -0,0 +1,197 @@
+    IHeaderDictionary Headers { get; }
+    long Length { get; }
+    string Name { get; }
+    string FileName { get; }
+    Stream OpenReadStream();
+    void CopyTo(Stream target);
+    Task CopyToAsync(Stream target, CancellationToken cancellationToken = null);
+  }
+
+When uploading files using model binding and the ``IFormFile`` interface, the action method can accept either a single ``IFormFile`` or an ``IEnumerable<IFormFile>`` (or ``List<IFormFile>``) representing several files. The following example loops through one or more uploaded files, saves them to the local file system, and then returns the total number and size of files uploaded.
+
+.. literalinclude:: file-uploads/sample/FileUploadSample/Controllers/UploadFilesController.cs
+  :language: c#
+  :emphasize-lines: 21,27,34
+
+Files uploaded using the ``IFormFile`` technique are buffered in memory on the web server before being processed. Inside the action method, the ``IFormFile`` contents are accessible as a stream. In addition to the local file system, files can be streamed to `Azure blob storage <https://azure.microsoft.com/en-us/documentation/articles/vs-storage-aspnet5-getting-started-blobs/>` or Entity Framework.
+
+To store binary file data in a database using Entity Framework, define a property of type ``byte[]`` on the entity:
+
+.. code-block:: c#
+
+  public class ApplicationUser : IdentityUser
+  {
+      public byte[] AvatarImage { get; set; }
+  }
+
+Specify a viewmodel property of type ``IFormFile``:
+
+.. code-block:: c#
+
+  public class RegisterViewModel
+  {
+      // other properties omitted
+
+      public IFormFile AvatarImage { get; set; }
+  }
+
+Copy the ``IFormFile`` to a stream and save it to the byte array:
+
+.. code-block:: c#
+
+  //
+  // POST: /Account/Register
+  [HttpPost]
+  [AllowAnonymous]
+  [ValidateAntiForgeryToken]
+  public async Task<IActionResult> Register(RegisterViewModel model)
+  {
+    ViewData["ReturnUrl"] = returnUrl;
+    if (ModelState.IsValid)
+    {
+        var user = new ApplicationUser { 
+          UserName = model.Email, 
+          Email = model.Email 
+        };
+        using (var memoryStream = new MemoryStream())
+        {
+            await model.AvatarImage.CopyToAsync(memoryStream);
+            user.AvatarImage = memoryStream.ToArray();
+        }
+    // additional logic omitted
+  }
+
+.. note:: Use caution when storing binary data in relational databases, as it can adversely impact performance.
+
+IFormFile and ViewModels
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition to accepting raw ``IFormFile`` parameters, action methods can accept viewmodel types with ``IFormFile`` properties. Consider the following ``ProfileEditViewModel``:
+
+.. literalinclude:: file-uploads/sample/FileUploadSample/ViewModels/ProfileEditViewModel.cs
+  :language: c#
+  :emphasize-lines: 11
+
+The following action method accepts this type and saves the associated ``AvatarFile`` property to the local file system:
```

Comment:
I wasn't able to get this working in the sample app, same problem with path combine as in the UploadFiles controller.